### PR TITLE
Speed up tree-sitter-language update script

### PR DESCRIPTION
**Pull request by Luni-4**
- Add an option to compute only new metrics
- Compute only the programming language associated to the `tree-sitter-language`

Thanks in advance for your review! :)

### CODE REVIEW

File: check-submodule.py [hunk 1]
```diff
@@ -66,23 +68,76 @@
...
+            # Dump json file of differences
+            dump_json_file(json_output, new_filename, self.compare_dir)
+
+            # Compute minimal tests
+            compute_minimal_tests(old_filename, new_filename, self.compare_dir)
+
+
 # Run a subprocess.
 def run_subprocess(cmd: str, *args: T.Union[str, pathlib.Path]) -> None:
     subprocess.run([cmd, *args])
```
Comment (marco-c) [inline check-submodule.py:138]: I wonder why git shows this as new. Could you move it to the top, so maybe it'll be easier for git to figure out it's still the same?

File: check-submodule.py [hunk 2]
```diff
@@ -66,23 +68,76 @@
...
+        self,
+        old_metrics: T.List[pathlib.Path],
+        new_metrics: T.List[pathlib.Path],
+        compare_dir: pathlib.Path,
+        max_workers: int,
+    ):
+        self.compare_dir = compare_dir
+        self.max_workers = max_workers
+
+        # Max number of filepaths in a sublist
```
Comment (marco-c) [inline check-submodule.py:84]: Nit: `file paths` or just `paths`

File: check-submodule.py [hunk 3]
```diff
@@ -66,23 +68,76 @@
...
+        n = math.ceil(len(old_metrics) / max_workers)
+
+        # Assign a certain number of filepaths to each worker
+        self.workers_filepaths = [
+            zip(old_metrics[i * n : (i + 1) * n], new_metrics[i * n : (i + 1) * n])
+            for i in range((len(old_metrics) + n - 1) // n)
```
Comment (marco-c) [inline check-submodule.py:90]: Nit: you can do this more easily if you use the third argument of range, something like `range(0, len(old_metrics), n)`.

>>> list(range(0, 100, 5))
[0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55, 60, 65, 70, 75, 80, 85, 90, 95]

File: check-submodule.py [hunk 4]
```diff
@@ -66,23 +68,76 @@
...
+        ]
+
+    # Run asynchronous comparisons between json files.
+    async def diff(self):
+        # Define the max number of coroutines used to compare json files
+        all_coroutines = asyncio.gather(
```
Comment (marco-c) [inline check-submodule.py:96]: Nit: don't define all_coroutines, just do `await asyncio.gather` directly

File: check-submodule.py [hunk 5]
```diff
@@ -66,23 +68,76 @@
...
+            *[
```
Comment (marco-c) [inline check-submodule.py:97]: Nit: try *(...) instead of *[...] to avoid creating a list

File: check-submodule.py [hunk 6]
```diff
@@ -43,6 +43,29 @@
 # Suffix for the directory containing the new metrics
 NEW_SUFFIX = "-new"
 
+# Extensions parsed by each tree-sitter-language
+EXTENSIONS = {
+    "tree-sitter-mozjs": ["*.js", "*.js2"],
```
Comment (marco-c) [inline check-submodule.py:50]: `.jsm` too

File: check-submodule.py [hunk 7]
```diff
@@ -43,6 +43,29 @@
...
+    "tree-sitter-javascript": ["*.js", "*.js2"],
+    "tree-sitter-tsx": ["*.tsx"],
+    "tree-sitter-typescript": ["*.ts", "*.jsw", "*.jsmw"],
+    "tree-sitter-java": ["*.java"],
+    "tree-sitter-rust": ["*.rs"],
+    "tree-sitter-cpp": [
```
Comment (marco-c) [inline check-submodule.py:56]: `cx` too

File: check-submodule.py [hunk 8]
```diff
@@ -203,22 +203,25 @@ def compute_metrics(args: argparse.Namespace) -> None:
     old_dir.mkdir(parents=True, exist_ok=True)
     new_dir.mkdir(parents=True, exist_ok=True)
 
-    # Git clone the chosen repository
-    # Note: no submodules repositories are accepted
-    print(f"Cloning {args.url} into {repo_dir}")
-    run_subprocess("git", "clone", "--depth=1", args.url, repo_dir)
+    # Skip whether only new metrics are requested
```
Comment (marco-c) [inline check-submodule.py:287]: `Skip when only new metrics are requested` or `Skip if only new metrics are requested`

File: check-submodule.py [hunk 9]
```diff
@@ -261,6 +264,14 @@ def main() -> None:
         "a tree-sitter-language update.",
     )
 
+    # Optional arguments
+    compute_metrics_cmd.add_argument(
+        "--new",
+        "-n",
+        action="store_true",
+        help="Compute only the metrics after the tree-sitter-language update",
```
Comment (marco-c) [inline check-submodule.py:353]: `Only compute the metrics after the tree-sitter-language update`

File: check-submodule.py [hunk 10]
```diff
@@ -261,6 +264,14 @@ def main() -> None:
         "a tree-sitter-language update.",
     )
 
+    # Optional arguments
+    compute_metrics_cmd.add_argument(
+        "--new",
```
Comment (marco-c) [inline check-submodule.py:350]: Let's call it `only-new` so it's more clear

**Comment by Luni-4** [comment]
@marco-c 

Let's address the asynchronous part in another PR, let's merge only these two commits for now

**Review by marco-c — APPROVED** [review]

**Pull request merged**